### PR TITLE
Chore: AI clean up

### PR DESCRIPTION
**Pull request by dcsmit**
## Description
Changed NPC to Visitor, changed AI to NPC.
Improved the target detection

## Setting up testing environment
1. In the project window navigate to Project/Assets/Scenes and open the "Mansion" scene.
2. Press Play.

#### Clean up
Criteria:
- [x] All NPC states should work as they should.

## Checklist
_These checks should always be true for your pull request_
- [x] My pull request is for one story/feature.
- [x] Every individual commit in this pull request is logical.
- [x] All code, documentation, commits and player seen texts are in English.
- [x] I have deleted unused / unnecessary code.
- [x] If my edits need a change in documentation, then I have updated the documentation.
- [x] All code is in correspondence with the code conventions.

### CODE REVIEW

File: Assets/Editor/NPC/SensesEditor.cs [hunk 1]
```diff
@@ -39,53 +39,53 @@ private void DrawFearReductionRange()
     }
 
     /// <summary>
-    /// Draws the AI's hearing radius on screen, which is the AI's auditory range.
+    /// Draws the npc's hearing radius on screen, which is the AI's auditory range.
```
Comment (Jeroenvn) [inline Assets/Editor/NPC/SensesEditor.cs]: "which is the AI's auditory range", dont you mean the npc's?

File: Assets/Scripts/NPC/NpcController.cs [hunk 2]
```diff
@@ -3,9 +3,9 @@
 using UnityEngine.AI;
 using UnityEngine.Events;
 
-public class AiController : MonoBehaviour
+public class NpcController : MonoBehaviour
 {
-    [Header("AI Settings")]
+    [Header("NPC Settings")]
     [Tooltip("The speed the AI will move when investigating."), Range(1f, 5f)]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/NpcController.cs]: "the speed the **AI** will...."

File: Assets/Scripts/NPC/NpcController.cs [hunk 3]
```diff
@@ -3,9 +3,9 @@
...
     public float InvestigatingSpeed = 2f;
     [Tooltip("The event that will be invoked when the ai changes state.")]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/NpcController.cs]: "**AI**"

File: Assets/Scripts/NPC/ReactionHandler.cs [hunk 4]
```diff
@@ -49,15 +49,15 @@ public class ReactionHandler : MonoBehaviour
 
     private IState _previousState;
     private Animator _animator;
-    private AiController _aiController;
+    private NpcController _aiController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/ReactionHandler.cs]: "_aiController"

File: Assets/Scripts/NPC/ReactionHandler.cs [hunk 5]
```diff
@@ -49,15 +49,15 @@ public class ReactionHandler : MonoBehaviour
...
 
     private void Awake()
     {
-        _aiController = GetComponent<AiController>();
+        _aiController = GetComponent<NpcController>();
         _animator = _reactionImage.GetComponent<Animator>();
         _aiController.OnStateChange.AddListener(OnStateChange);
 
-        if(_aiController.TryGetComponent(out NpcController npcController))
+        if(_aiController.TryGetComponent(out VisitorController npcController))
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/ReactionHandler.cs]: npcController should be visitorController

File: Assets/Scripts/NPC/Realtor/RealtorSenses.cs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        base.ClearDetectedTargets();
+        foreach(VisitorController visitor in SoothedVisitors)
+        {
+            visitor.SeenByRealtor = false;
+        }
+        SoothedVisitors.Clear();
+    }
+
+    private IEnumerator DecreaseNpcFear()
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Realtor/RealtorSenses.cs:85]: Shouldn't this be Decrease**Visitor**Fear()?

File: Assets/Scripts/NPC/Senses/BaseSenses.cs [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    public float SightRange = 20f;
+
+    [Header("Auditory Settings")]
+    [Tooltip("The distance that the AI can hear."), Range(0, 50)]
+    public float AuditoryRange = 15f;
+    
+    [HideInInspector]
+    public Dictionary<ObservableObject, DetectedProperties> DetectedObjects = new();
+    [HideInInspector]
+    public List<VisitorController> DetectedNpcs = new();
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Senses/BaseSenses.cs]: Shouldnt this be Detected**Visitors**

File: Assets/Scripts/NPC/Senses/BaseSenses.cs [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    [Tooltip("The obstacle layers that block the entity's vision."), SerializeField]
+    protected LayerMask _obstacleMask;
+    [Header("Sight Settings")]
+    [Tooltip("The angle of the AI's field of view."), Range(0, 360)]
+    public float FieldOfViewAngle = 110f;
+    [Tooltip("The distance that the AI can see."), Range(0, 50)]
+    public float SightRange = 20f;
+
+    [Header("Auditory Settings")]
+    [Tooltip("The distance that the AI can hear."), Range(0, 50)]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Senses/BaseSenses.cs]: ai

File: Assets/Scripts/NPC/Senses/BaseSenses.cs [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    [Tooltip("The transform that the entity uses to detect targets."), SerializeField]
+    public Transform _headTransform;
+    [Tooltip("The target layers that the entity detects."), SerializeField]
+    protected LayerMask _targetMask;
+    [Tooltip("The obstacle layers that block the entity's vision."), SerializeField]
+    protected LayerMask _obstacleMask;
+    [Header("Sight Settings")]
+    [Tooltip("The angle of the AI's field of view."), Range(0, 360)]
+    public float FieldOfViewAngle = 110f;
+    [Tooltip("The distance that the AI can see."), Range(0, 50)]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Senses/BaseSenses.cs]: ai

File: Assets/Scripts/NPC/Senses/BaseSenses.cs [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+{
+    [Header("NPC Senses Settings")]
+    [Tooltip("The transform that the entity uses to detect targets."), SerializeField]
+    public Transform _headTransform;
+    [Tooltip("The target layers that the entity detects."), SerializeField]
+    protected LayerMask _targetMask;
+    [Tooltip("The obstacle layers that block the entity's vision."), SerializeField]
+    protected LayerMask _obstacleMask;
+    [Header("Sight Settings")]
+    [Tooltip("The angle of the AI's field of view."), Range(0, 360)]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Senses/BaseSenses.cs]: ai

File: Assets/Scripts/NPC/Senses/BaseSenses.cs [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    protected virtual void HandleTargets(Collider[] targetsInDetectionRadius)
+    {
+        ClearDetectedTargets();
+
+        for (int i = 0; i < targetsInDetectionRadius.Length; i++)
+        {
+            Collider target = targetsInDetectionRadius[i];
+            bool isNpc = target.TryGetComponent<VisitorController>(out var npc);
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Senses/BaseSenses.cs]: isNpc -> isVisitor

File: Assets/Scripts/NPC/States/IdleState.cs [hunk 12]
```diff
@@ -3,9 +3,9 @@
 
 public class IdleState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/IdleState.cs]: _npcController -> _visitorController

File: Assets/Scripts/NPC/States/IdleState.cs [hunk 13]
```diff
@@ -3,9 +3,9 @@
...
 
-    public IdleState(NpcController npcController)
+    public IdleState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/IdleState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/InvestigateState.cs [hunk 14]
```diff
@@ -3,13 +3,13 @@
 
 public class InvestigateState : IState
 {
-    private readonly AiController _aiController;
+    private readonly NpcController _aiController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/InvestigateState.cs]: _aiController -> _npcController

File: Assets/Scripts/NPC/States/InvestigateState.cs [hunk 15]
```diff
@@ -3,13 +3,13 @@
...
 
     private Collider _investigateTargetCollider;
 
     private IState _stateToReturnTo;
 
-    public InvestigateState(AiController aiController, IState stateToReturnTo)
+    public InvestigateState(NpcController aiController, IState stateToReturnTo)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/InvestigateState.cs]: ai -> npc

File: Assets/Scripts/NPC/States/PanickedState.cs [hunk 16]
```diff
@@ -3,9 +3,9 @@
 
 public class PanickedState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/PanickedState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/PanickedState.cs [hunk 17]
```diff
@@ -3,9 +3,9 @@
...
 
-    public PanickedState(NpcController npcController)
+    public PanickedState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/PanickedState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/PhobiaState.cs [hunk 18]
```diff
@@ -3,9 +3,9 @@
 
 public class PhobiaState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/PhobiaState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/PhobiaState.cs [hunk 19]
```diff
@@ -3,9 +3,9 @@
...
 
-    public PhobiaState(NpcController npcController)
+    public PhobiaState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/PhobiaState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/RandomRoamState.cs [hunk 20]
```diff
@@ -4,9 +4,9 @@
 
 public class RandomRoamState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/RandomRoamState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/RandomRoamState.cs [hunk 21]
```diff
@@ -4,9 +4,9 @@
...
     
-    public RandomRoamState(NpcController npcController)
+    public RandomRoamState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/RandomRoamState.cs]: npc -> visitor

File: Assets/Scripts/NPC/States/RoamState.cs [hunk 22]
```diff
@@ -4,9 +4,9 @@
 
 public class RoamState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/RoamState.cs]: npc

File: Assets/Scripts/NPC/States/RoamState.cs [hunk 23]
```diff
@@ -4,9 +4,9 @@
...
 
-    public RoamState(NpcController npcController)
+    public RoamState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/RoamState.cs]: npc

File: Assets/Scripts/NPC/States/ScaredState.cs [hunk 24]
```diff
@@ -3,9 +3,9 @@
 
 public class ScaredState : IState
 {
-    private readonly NpcController _npcController;
+    private readonly VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/ScaredState.cs]: npc

File: Assets/Scripts/NPC/States/ScaredState.cs [hunk 25]
```diff
@@ -3,9 +3,9 @@
...
 
-    public ScaredState(NpcController npcController)
+    public ScaredState(VisitorController npcController)
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/States/ScaredState.cs]: npc

File: Assets/Scripts/NPC/Visitor/FearBarController.cs [hunk 26]
```diff
@@ -6,12 +6,12 @@ public class FearBarController : MonoBehaviour
     private Slider _fearBar;
 
     private readonly float _maxFear = 100;
-    private NpcController _npcController;
+    private VisitorController _npcController;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Visitor/FearBarController.cs]: npc

File: Assets/Scripts/NPC/Visitor/FearHandler.cs [hunk 27]
```diff
@@ -31,17 +31,17 @@ public class FearHandler : MonoBehaviour
     [Tooltip("Amount of added fear needed for NPC to get scared"), SerializeField]
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Visitor/FearHandler.cs]: npc

File: Assets/Scripts/NPC/Visitor/FearHandler.cs [hunk 28]
```diff
@@ -31,17 +31,17 @@ public class FearHandler : MonoBehaviour
...
     private float _scaredThreshold = 22f;
 
-    private NpcController _npcController;
-    private NpcSenses _npcSenses;
+    private VisitorController _npcController;
+    private VisitorSenses _npcSenses;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Visitor/FearHandler.cs]: npc

File: Assets/Scripts/NPC/Visitor/VisitorManager.cs [hunk 29]
```diff
@@ -0,0 +1,43 @@
...
+public class VisitorManager : MonoBehaviour
+{
+    public delegate void VisitorsLeftChanged(int value);
+    public event VisitorsLeftChanged OnVisitorsLeftChanged;
+
+    private VisitorController[] _visitors;
+    private int _scaredVisitors;
+
+    [Header("References")]
+    public GameObject NpcCollection;
```
Comment (Jeroenvn) [inline Assets/Scripts/NPC/Visitor/VisitorManager.cs]: is it still npc collection or is it now visitor collection?

**Review by Jeroenvn — CHANGES_REQUESTED** [review]
Everything works! But there are still some visitors disquised as npc's running around

File: Assets/Scripts/NPC/Realtor/RealtorSenses.cs (hunk 6 above)
Comment (dcsmit) [reply Assets/Scripts/NPC/Realtor/RealtorSenses.cs:85]: That one is going to get removed anyways

**Review by Jeroenvn — APPROVED** [review]
Looks good!

**Pull request merged**